it/uk/gov/hmrc/agentmappingfrontend/stubs/AuthStub.scala

```diff
@@ -29,4 +29,21 @@ object AuthStub {
     stubFor(get(urlEqualTo(user.enrolmentsLink)).willReturn(aResponse().withStatus(200).withBody(
       s"""|[{"key":"IR-SA-AGENT","identifiers":[{"key":"IrAgentReference","value":"${user.saAgentReference.get}"}],"state":"$state"}]""".stripMargin)))
   }
+
+  def passcodeAuthorisationSucceeds(regime: String = "agent-mapping", otacToken: String = "dummy-otac-token"): Seq[(String, String)] = {
+    stubPasscodeAuthorisation(regime, 200)
+
+    Seq(SessionKeys.otacToken -> otacToken)
+  }
+
+  def passcodeAuthorisationFails(regime: String = "agent-mapping"): Unit = {
+    stubPasscodeAuthorisation(regime, 404)
```

Comment:
Does the real service really return a 404 when passcode authorisation fails (not a 401 or a 403)?